scx_bpfland: rely on Topology to classify CPU types

arighi opened the pull request:
Integrate in Topology the concept of "Turbo" CPUs (to represent the turbo-boosted cores), refactor scx_bpfland to use simply rely on Topology to classify the CPUs in Turbo, Big and Little.

@hodgesds let me know if you see any problem with the new core type "Turbo" or if you have a better idea to represent it in struct Cpu (this is more of a RFC).

htejun commented on rust/scx_utils/src/topology.rs:
```diff
@@ -97,6 +97,7 @@ lazy_static::lazy_static! {
 #[derive(Debug, Clone, Eq, Hash, PartialEq)]
 pub enum CoreType {
+    Turbo,
     Big,
```
Would it make sense to make `Turbo` a variant of `Big`? ie. so that it's `Big { turbo: bool }`? That way, the hierarchy is encoded in the type system and people can use `match` more naturally rather than using `is_*()` helpers.

arighi replied:
Hm... that seems really nice, will check that!

hodgesds commented:
I think this should work well, don't have access to a turbo type Intel CPU machine 😄

arighi commented:
Added @htejun's  idea (turbo as a property of Big) and also added the special strings `all` and `none` to inizialize a cpumask using `Cpumask::from_str()`. I need this in scx_bpfland to make the code cleaner, but they could be useful for other cases as well.

hodgesds reviewed (approved):
This is much nicer, thanks for cleaning it up!
